pkg/controller/configmap/configmap_controller.go

```diff
@@ -630,7 +650,46 @@ func (r *ReconcileConfigMap) isNcpDeploymentChanged(ncpReplicas int32) (bool, er
 	}
 	prevImage := ncpDeployment.Spec.Template.Spec.Containers[0].Image
 	currImage := os.Getenv(operatortypes.NcpImageEnv)
-	if prevImage != currImage || ncpReplicas != *ncpDeployment.Spec.Replicas {
+
+	prevTolerations := ncpDeployment.Spec.Template.Spec.Tolerations
+	isTolerationsEqual := true
+	if ncpTolerations != nil {
+		isTolerationsEqual = reflect.DeepEqual(prevTolerations, *ncpTolerations)
+	}
```

Comment:
What if prevTolerations!=nil and ncpTolerations==nil?

Reply:
ncpTolerations should not be nil, it's not expected. as we discussed, since DeepEqual could check nil inside, we just pass ncpTolerations instead of *ncpTolerations